### DISCUSSION TITLE
Where can I place a script that will be global

### DISCUSSION THREAD

**Discussion by scottw12** [vanjarosoftware/Vanjaro.Platform#1038] [Q&A] [answered]
I need to insert some JavaScript that will be global to all pages irregardless of template. This will be updated occasionally by the developers and is not needed to be updated by editors.    We have some advertising and analytics tags we need to add to every page in the site.  I would rather do it at source (ie. like default.aspx) level than have to modify several templates, global blocks.

**Comment by scottw12** [comment]
Another thing I want to do is place a "back-to-top" icon on the pages and provide a floating button to navigate to the top of the page.  
I use the following code to monitor window position and make the button visible;
 <script>
                 ```
   function backToTop() {
                        /* Check show or hide the button [back-to-top] */
                        $(window).scroll(function() {
                            if ($(this).scrollTop() > 50) {
                                $('#back-to-top').fadeIn();
                            } else {
                                $('#back-to-top').fadeOut();
                            }
                        });

                        // scroll body to 0px on click
                        $('#back-to-top').click(function() {
                            $('body,html').animate({
                                scrollTop: 0
                            }, 800);
                            return false;
                        });

                    }
```
                </script>

**Comment by meetmandeep** [comment, marked as answer]
Modifying default.aspx is never recommended. 

You can easily add your own Global Scripts in Header tag from Menu > Settings > SEO & Site Search > Site > Advanced > HTML Page Header Tags

**Comment by scottw12** [comment]
Soo there is no way to properly add script to the bottom of every page in one place?   BTW I am still debugging to find the exact scenario but it seems that Live Forms  breaks the custom code added to the same page... I will get back to that later.

**Reply by meetmandeep** [reply]
HTML Page Header Tags is GLOBAL; you're not doing this on page by page basis. That's for the entire site. It's no different then modifying default.aspx. 

Default.aspx modification applies to all sites/portals. The Page Header Tags would apply to a single site only (the recommended approach)

**Reply by scottw12** [reply]
I know that page header tags is global...but I am a stickler with the layout of pages etc. I was taught that all supporting script and script libraries should reside at the bottom of the form and not in the header for optimization. I will go with what I have available.

**Reply by meetmandeep** [reply]
Add it as an enhancement request to Issues.

**Reply by scottw12** [reply]
A word of Warning!!!   If you add script into the SEO header and that script is malformed you will disable the entire site.
I had a malformed <script><script> (instead of <script></script>) and broke the entire site. I had to go to the sql table PortalSettings and manually remove the offending code to fix things.
